clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/BackgroundEventHandler.java

```diff
@@ -42,6 +47,8 @@ public BackgroundEventHandler(final Queue<BackgroundEvent> backgroundEventQueue)
      */
     public void add(BackgroundEvent event) {
         Objects.requireNonNull(event, "BackgroundEvent provided to add must be non-null");
+        event.setEnqueuedMs(System.currentTimeMillis());
```

Comment:
We need to add a `Time` variable to this class so that we can do this instead:

```suggestion
        event.setEnqueuedMs(time.milliseconds());
```